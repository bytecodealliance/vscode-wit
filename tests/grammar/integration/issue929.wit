package foo:bar4;

interface f {
    resource fd;
}

interface utils {
    use f.{fd};
    my-func: func() -> own<fd>;
}

world test {
    import f;
    export f;
    export utils;
}
