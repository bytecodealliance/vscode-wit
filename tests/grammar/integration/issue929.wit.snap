>package foo:bar4;
#^^^^^^^ source.wit meta.package-decl.wit storage.modifier.package-decl.wit
#       ^ source.wit meta.package-decl.wit
#        ^^^ source.wit meta.package-decl.wit meta.id.package-decl.wit meta.package-identifier.wit entity.name.namespace.package-identifier.wit entity.name.type.id.wit
#           ^ source.wit meta.package-decl.wit meta.id.package-decl.wit meta.package-identifier.wit keyword.operator.namespace.package-identifier.wit
#            ^^^^ source.wit meta.package-decl.wit meta.id.package-decl.wit meta.package-identifier.wit entity.name.type.package-identifier.wit entity.name.type.id.wit
#                ^ source.wit meta.package-decl.wit meta.id.package-decl.wit meta.package-identifier.wit entity.name.type.package-identifier.wit
>
>interface f {
#^^^^^^^^^ source.wit meta.interface-item.wit keyword.declaration.interface.interface-item.wit storage.type.wit
#         ^ source.wit meta.interface-item.wit
#          ^ source.wit meta.interface-item.wit entity.name.type.id.interface-item.wit
#           ^ source.wit meta.interface-item.wit
#            ^ source.wit meta.interface-item.wit punctuation.brackets.curly.begin.wit
>    resource fd;
#^^^^ source.wit meta.interface-item.wit meta.whitespace.wit
#    ^^^^^^^^ source.wit meta.interface-item.wit meta.resource-item.wit keyword.other.resource.wit
#            ^ source.wit meta.interface-item.wit meta.resource-item.wit
#             ^^ source.wit meta.interface-item.wit meta.resource-item.wit entity.name.type.id.resource.wit
#               ^ source.wit meta.interface-item.wit meta.resource-item.wit
>}
#^ source.wit meta.interface-item.wit punctuation.brackets.curly.end.wit
>
>interface utils {
#^^^^^^^^^ source.wit meta.interface-item.wit keyword.declaration.interface.interface-item.wit storage.type.wit
#         ^ source.wit meta.interface-item.wit
#          ^^^^^ source.wit meta.interface-item.wit entity.name.type.id.interface-item.wit
#               ^ source.wit meta.interface-item.wit
#                ^ source.wit meta.interface-item.wit punctuation.brackets.curly.begin.wit
>    use f.{fd};
#^^^^ source.wit meta.interface-item.wit meta.whitespace.wit
#    ^^^ source.wit meta.interface-item.wit meta.use-item.wit keyword.other.use.use-item.wit
#       ^ source.wit meta.interface-item.wit meta.use-item.wit
#        ^ source.wit meta.interface-item.wit meta.use-item.wit entity.name.namespace.id.use-path.wit
#         ^ source.wit meta.interface-item.wit meta.use-item.wit keyword.operator.namespace-separator.use-item.wit
#          ^ source.wit meta.interface-item.wit meta.use-item.wit punctuation.brackets.curly.begin.wit
#           ^^ source.wit meta.interface-item.wit meta.use-item.wit entity.name.type.declaration.use-names-item.use-item.wit
#             ^ source.wit meta.interface-item.wit meta.use-item.wit punctuation.brackets.curly.end.wit
#              ^ source.wit meta.interface-item.wit
>    my-func: func() -> own<fd>;
#^^^^ source.wit meta.interface-item.wit meta.whitespace.wit
#    ^^ source.wit meta.interface-item.wit meta.func-item.wit entity.name.function.id.func-item.wit meta.word.wit
#      ^ source.wit meta.interface-item.wit meta.func-item.wit entity.name.function.id.func-item.wit meta.word-separator.wit
#       ^^^^ source.wit meta.interface-item.wit meta.func-item.wit entity.name.function.id.func-item.wit meta.word.wit
#           ^ source.wit meta.interface-item.wit meta.func-item.wit keyword.operator.key-value.wit
#            ^ source.wit meta.interface-item.wit meta.func-item.wit meta.whitespace.wit
#             ^^^^ source.wit meta.interface-item.wit meta.func-item.wit meta.function.wit keyword.other.func.func-type.wit
#                 ^ source.wit meta.interface-item.wit meta.func-item.wit meta.function.wit meta.param-list.wit punctuation.brackets.round.begin.wit
#                  ^ source.wit meta.interface-item.wit meta.func-item.wit meta.function.wit meta.param-list.wit punctuation.brackets.round.end.wit
#                   ^ source.wit meta.interface-item.wit meta.func-item.wit meta.function.wit meta.whitespace.wit
#                    ^^ source.wit meta.interface-item.wit meta.func-item.wit meta.function.wit meta.result-list.wit keyword.operator.arrow.skinny.wit
#                      ^ source.wit meta.interface-item.wit meta.func-item.wit meta.function.wit meta.result-list.wit meta.whitespace.wit
#                       ^^^ source.wit meta.interface-item.wit meta.func-item.wit meta.function.wit meta.result-list.wit entity.name.type.id.wit
#                          ^ source.wit meta.interface-item.wit meta.func-item.wit meta.function.wit meta.result-list.wit
#                           ^^ source.wit meta.interface-item.wit meta.func-item.wit meta.function.wit meta.result-list.wit entity.name.type.id.wit
#                             ^^ source.wit meta.interface-item.wit meta.func-item.wit meta.function.wit meta.result-list.wit
>}
#^ source.wit meta.interface-item.wit punctuation.brackets.curly.end.wit
>
>world test {
#^^^^^ source.wit meta.world-item.wit keyword.declaration.world.world-item.wit storage.type.wit
#     ^ source.wit meta.world-item.wit
#      ^^^^ source.wit meta.world-item.wit entity.name.type.id.world-item.wit
#          ^ source.wit meta.world-item.wit
#           ^ source.wit meta.world-item.wit punctuation.brackets.curly.begin.wit
>    import f;
#^^^^ source.wit meta.world-item.wit meta.whitespace.wit
#    ^^^^^^ source.wit meta.world-item.wit meta.import-item.wit keyword.control.import.import-item.wit
#          ^ source.wit meta.world-item.wit meta.import-item.wit
#           ^ source.wit meta.world-item.wit meta.import-item.wit meta.id.import-item.wit variable.other.id.import-item.wit
#            ^ source.wit meta.world-item.wit meta.import-item.wit meta.id.import-item.wit
>    export f;
#^^^^ source.wit meta.world-item.wit meta.whitespace.wit
#    ^^^^^^ source.wit meta.world-item.wit meta.export-item.wit keyword.control.export.export-item.wit
#          ^ source.wit meta.world-item.wit meta.export-item.wit
#           ^ source.wit meta.world-item.wit meta.export-item.wit meta.id.export-item.wit variable.other.constant.id.export-item.wit
#            ^ source.wit meta.world-item.wit meta.export-item.wit meta.id.export-item.wit
>    export utils;
#^^^^ source.wit meta.world-item.wit meta.whitespace.wit
#    ^^^^^^ source.wit meta.world-item.wit meta.export-item.wit keyword.control.export.export-item.wit
#          ^ source.wit meta.world-item.wit meta.export-item.wit
#           ^^^^^ source.wit meta.world-item.wit meta.export-item.wit meta.id.export-item.wit variable.other.constant.id.export-item.wit
#                ^ source.wit meta.world-item.wit meta.export-item.wit meta.id.export-item.wit
>}
#^ source.wit meta.world-item.wit punctuation.brackets.curly.end.wit
>